irohad/ametsuchi/impl/flat_file/flat_file.cpp

```diff
@@ -102,22 +101,26 @@ nonstd::optional<Identifier> check_consistency(const std::string &dump_dir) {
 
   auto n = static_cast<uint32_t>(status);
   tmp_id++;
+
+  // get last available identifier
+  auto last = tmp_id;
```

Comment:
This loop seems more clear if you do it in two loops: find non-consecutive file name, and remove files after it, including itself.
```C++
Identifier tmp_id = 2;
while (tmp_id < n and id_to_name(tmp_id - 1) == namelist[tmp_id]->d_name) {
  ++tmp_id;
}
for (auto j = tmp_id; j < n; ++j) {
  remove(dump_dir, namelist[j]->d_name);
}
...
return tmp_id - 2;
```

Reply:
why start with identifier equals 2?

Reply:
ah, understand